From 11e6354aeb896df52f36d9a6dfec3f6ca18b266e Mon Sep 17 00:00:00 2001
From: sendaoYan <yansendao.ysd@alibaba-inc.com>
Date: Fri, 26 Jan 2024 16:45:36 +0800
Subject: [PATCH] fix bugs: const MAX_FILTER_KIND_SIZE outside the valid range

fix bugs: src/Filter.h:60:14: error: integer value 2 is outside the valid range of values [0, 1] for the enumeration type 'FilterKind' [-Wenum-constexpr-conversion]; https://github.com/csmith-project/csmith/issues/163

Fixes: #163
Signed-off-by: sendaoYan <yansendao.ysd@alibaba-inc.com>
---
 src/Filter.h | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/src/Filter.h b/src/Filter.h
index 0482d8b48..a1600e95f 100644
--- a/src/Filter.h
+++ b/src/Filter.h
@@ -35,10 +35,9 @@
 enum FilterKind {
 	fDefault,
 	fDFS,
+	MAX_FILTER_KIND_SIZE,
 };
 
-#define MAX_FILTER_KIND_SIZE ((FilterKind) (fDFS + 1))
-
 // Filter base class
 class Filter
 {
